Resources/Prototypes/Entities/Structures/Power/Generation/supermatter.yml

```diff
@@ -0,0 +1,51 @@
+- type: entity
+  id: supermatter
+  name: Supermatter
+  description: A strangely translucent and iridescent crystal.
+  components:
+    - type: Supermatter
+      whitelist:
+        tags:
+          - EmitterBolt
+        components:
+          - Body
+          - Item
+    - type: Damageable
+      damageContainer: Inorganic
+      damageModifierSet: Glass
+    - type: Physics
+      bodyType: Dynamic
+    - type: Fixtures
+      fixtures:
+      - shape:
+          !type:PhysShapeAabb
+            bounds: "-0.25,-0.25,0.25,0.25"
+        mass: 250
+        layer:
+          - Impassable
+          - MobImpassable
+          - VaultImpassable
+          - Opaque
+        mask:
+          - Impassable
+          - MobImpassable
+          - VaultImpassable
+          - SmallImpassable
```

Comment:
I'd recommend Impassable for the mask and Opaque + impassable + vaultimpassable + smallimpassable for the layer.